File: packages/sdk-php/src/Exception/ZenvioApiException.php
<?php

namespace Zenvio\Exception;

use Exception;

class ZenvioApiException extends Exception
{
    public function __construct(
        public readonly int $statusCode,
        public readonly string $responseBody,
        ?Exception $previous = null
    ) {
        parent::__construct(
            "Zenvio API error {$statusCode}: {$responseBody}",
            0,
            $previous
        );
    }
}

File: packages/sdk-dotnet/src/Zenvio/ZenvioApiException.cs
namespace Zenvio;

public class ZenvioApiException : Exception
{
    public int StatusCode { get; }
    public string ResponseBody { get; }

    public ZenvioApiException(int statusCode, string responseBody)
        : base($"Zenvio API error {statusCode}: {responseBody}")
    {
        StatusCode = statusCode;
        ResponseBody = responseBody;
    }
}
